ant-node/src/replication.rs

```diff
@@ -174,4 +176,61 @@ impl Node {
             );
         });
     }
+
+    // To fetch a received fresh record replication
+    pub(crate) fn fresh_replicate_to_fetch(
+        &self,
+        holder: NetworkAddress,
+        keys: Vec<(NetworkAddress, ValidationType, Option<ProofOfPayment>)>,
+    ) {
+        let node = self.clone();
+        let _handle = spawn(async move {
+            let mut new_keys = vec![];
+            for (addr, val_type, payment) in keys {
+                if let Some(payment) = payment {
+                    // Payment must be valid
+                    match node
+                        .payment_for_us_exists_and_is_still_valid(&addr, payment)
+                        .await
+                    {
+                        Ok(_) => {}
+                        Err(err) => {
+                            info!("ProofOfPayment of {addr:?} is invalid with error {err:?}");
+                            continue;
+                        }
+                    }
+                } else {
+                    // Must have existing copy
+                    match node
+                        .validate_key_and_existence(&addr, &addr.to_record_key())
+                        .await
+                    {
+                        Ok(true) => {}
+                        Ok(false) => {
+                            info!(
+                                "Received a fresh update against a non-existing record of {addr:?}"
+                            );
+                            continue;
+                        }
+                        Err(err) => {
+                            info!("Failed to verify the local existence of {addr:?} with error {err:?}");
+                            continue;
+                        }
```

Comment:
Do we deal with split record case here? 
Say I have a different GraphEntry or ScratchPad at the same address here, what happens?

Reply:
here don't need to.
it's when the copy fetched, then to handle the split case.